src/python/pants/build_graph/target.py

```diff
@@ -473,7 +473,10 @@ def transitive_invalidation_hash(self, fingerprint_strategy=None, depth=0):
       def dep_hash_iter():
         for dep in self.dependencies:
           try:
-            dep_hash = dep.transitive_invalidation_hash(fingerprint_strategy, depth=depth+1)
+            if getattr(self, 'strict_deps', None):
+              dep_hash = dep.invalidation_hash(fingerprint_strategy)
```

Comment:
that sounds reasonable. so I will add a new state here to invalidate direct dependees. I will change the parameter type from boolean to constant integers as now there are 3 cases.